iroh/src/client.rs

```diff
@@ -44,9 +56,71 @@ where
 {
     /// Create a new high-level client to a Iroh node from the low-level RPC client.
     pub fn new(rpc: RpcClient<ProviderService, C>) -> Self {
-        Self { rpc }
+        Self {
+            node: NodeClient { rpc: rpc.clone() },
+            blobs: BlobsClient { rpc: rpc.clone() },
+            docs: DocsClient { rpc },
+        }
+    }
+}
+
+/// Iroh node client.
+#[derive(Debug, Clone)]
+pub struct NodeClient<C> {
+    rpc: RpcClient<ProviderService, C>,
+}
+
+impl<C> NodeClient<C>
+where
+    C: ServiceConnection<ProviderService>,
+{
+    /// Get statistics of the running node.
+    pub async fn stats(&self) -> Result<HashMap<String, CounterStats>> {
+        let res = self.rpc.rpc(NodeStatsRequest {}).await??;
+        Ok(res.stats)
+    }
+
+    /// Get information about the different connections we have made
+    pub async fn connections(&self) -> Result<impl Stream<Item = Result<ConnectionInfo>>> {
+        let stream = self.rpc.server_streaming(NodeConnectionsRequest {}).await?;
+        Ok(flatten(stream).map_ok(|res| res.conn_info))
     }
 
+    /// Get connection information about a node
+    pub async fn connection_info(&self, node_id: PublicKey) -> Result<Option<ConnectionInfo>> {
+        let NodeConnectionInfoResponse { conn_info } = self
+            .rpc
+            .rpc(NodeConnectionInfoRequest { node_id })
+            .await??;
+        Ok(conn_info)
+    }
+
+    /// Get status information about a node
+    pub async fn status(&self) -> Result<NodeStatusResponse> {
+        let response = self.rpc.rpc(NodeStatusRequest).await?;
+        Ok(response)
+    }
+
+    /// Shutdown the node.
+    ///
+    /// If `force` is true, the node will be killed instantly without waiting for things to
+    /// shutdown gracefully.
+    pub async fn shutdown(&self, force: bool) -> Result<()> {
+        self.rpc.rpc(NodeShutdownRequest { force }).await?;
+        Ok(())
+    }
+}
+
+/// Iroh docs client.
+#[derive(Debug, Clone)]
+pub struct DocsClient<C> {
+    rpc: RpcClient<ProviderService, C>,
+}
+
+impl<C> DocsClient<C>
+where
+    C: ServiceConnection<ProviderService>,
+{
     /// Create a new document author.
     pub async fn create_author(&self) -> Result<AuthorId> {
```

Comment:
I wonder if we want to add another layer (or potentially to the top layer)

- `iroh.docs.authors.new()` or 
- `iroh.authors.new()`

Reply:
Changed it to `iroh.authors.create` for now.